### PR TITLE
fix(bridge): truncate derived session titles on grapheme boundaries

### PR DESCRIPTION
**Pull request by 0xfandom**
### Problem

The bridge derives a session title from the user's first REPL message and PATCHes it to the claude.ai backend, where it's JSON-serialized and UTF-8-encoded for the remote/mobile session list.

`deriveTitle` truncates with `flat.slice(0, TITLE_MAX_LEN - 1)`, a **UTF-16 code-unit slice**. When an emoji or astral-plane character straddles the cut, the slice keeps its high surrogate and drops the low one, leaving a lone surrogate. Over the UTF-8 wire that becomes the U+FFFD replacement character — the session title shows mojibake.

Reproduced against the exact slice logic with a first message of `'a'.repeat(48) + '😀😀😀 fix the login bug'` (the 😀 lands on the 50-char boundary):

```
title.isWellFormed() = false           // lone high surrogate U+D83D
wire body            = {"title":"aaa…aaa\ud83d…"}
UTF-8 roundtrip      = "aaa…aaa�…"      // U+FFFD
```

### Fix

Route through `truncateToWidth`, the grapheme-safe helper that the parallel derivation `deriveSessionTitle` in `bridgeMain.ts` already uses for the identical purpose. It segments on grapheme boundaries, so it never splits a surrogate pair, and appends `…` on truncation — matching the original intent.

One-line source change plus a regression test covering the emoji-boundary case, long-title ellipsis, short-title passthrough, empty message, and whitespace collapse.

Same class as the recently merged byte/UTF-8 truncation fixes (#1918, #1960).

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved title truncation to be grapheme/emoji-safe, avoiding broken characters and lone-surrogate artifacts.
  * Ensures truncated output uses an ellipsis when needed, stays within the defined character-length limit, and avoids introducing replacement symbols.
  * Normalizes multiline titles into a single-line result and omits titles when input is empty.

* **Tests**
  * Added coverage for emoji boundaries, surrogate safety, UTF-8 round-trip validity, character-count caps (including CJK), ellipsis behavior, and multiline/empty cases.

### CODE REVIEW

File: src/bridge/initReplBridge.titleTruncation.test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+// PATCHes it to the claude.ai backend, where it is JSON-serialized and
+// UTF-8-encoded. A raw `.slice(0, N)` cut at a UTF-16 code-unit index can split
+// an emoji or astral-plane character's surrogate pair, leaving a lone surrogate
+// that goes over the wire as the U+FFFD replacement character.
+
+// A high surrogate not followed by a low one (or vice versa) is an unpaired
+// code unit — exactly what a mid-pair slice leaves behind.
+const LONE_SURROGATE =
+  /[\uD800-\uDBFF](?![\uDC00-\uDFFF])|(?<![\uD800-\uDBFF])[\uDC00-\uDFFF]/
+
```
[the remark this answers is not in the document]
Comment (0xfandom) [reply src/bridge/initReplBridge.titleTruncation.test.ts:17]: Good point — the source regex deliberately avoids lookbehinds for YARR/JSC, so the test shouldn't reintroduce one. Switched to the leading `(?:^|[^\uD800-\uDBFF])` alternation in 723d5109. Still passes.

**Review by jatmn — CHANGES_REQUESTED** [review]
I found issues that need to be addressed before this is ready.

## Findings

- [P1] Preserve a character/transport bound instead of using terminal display width
  `src/bridge/initReplBridge.ts:580`
  `TITLE_MAX_LEN` is the 50-character limit for the session-title API field, but `truncateToWidth` measures terminal columns. This both truncates valid wide-script titles early (for example, 30 `你` characters become 24 plus `…`) and removes the effective cap for zero-width graphemes: `deriveTitle('\u200B'.repeat(100000))` returns all 100,000 code units because its display width is zero. The result is logged and PATCHed as the remote session title, so ordinary CJK titles lose content while a pasted or injected combining/zero-width-heavy prompt can produce an arbitrarily large request. Please use a grapheme-safe truncator that also enforces this field's character or API-size limit, and cover both wide and zero-width inputs.

**Comment by 0xfandom** [comment]
Rebased onto current main. Good catch — `truncateToWidth` measures terminal columns, but `TITLE_MAX_LEN` bounds the API field in characters, so I traded one bug for two: 30 CJK characters truncated to 24, and 100k zero-width code units passed through uncapped (confirmed both). Now walks graphemes and accumulates against code-unit length, which keeps the surrogate pair intact — the original defect — while enforcing the character bound. Regressions added for both directions.

**Review by jatmn — APPROVED** [review]
@kevincodex1 LGTM

**Review by kevincodex1 — APPROVED** [review]
